WordPressCom-Stats-iOS/WPStatsService.m

```diff
@@ -173,6 +204,90 @@ - (void)retrieveAllStatsForDate:(NSDate *)date
 }
 
 
+- (void)retrieveInsightsStatsWithAllTimeStatsCompletionHandler:(StatsAllTimeCompletion)allTimeCompletion
+                                     insightsCompletionHandler:(StatsInsightsCompletion)insightsCompletion
+                                 todaySummaryCompletionHandler:(StatsSummaryCompletion)todaySummaryCompletion
+                                                 progressBlock:(void (^)(NSUInteger numberOfFinishedOperations, NSUInteger totalNumberOfOperations)) progressBlock
+                                   andOverallCompletionHandler:(void (^)())overallCompletionHandler
+{
+    NSString *cacheKey = @"BatchInsights";
+    NSString *allTimeCacheKey = @"AllTime";
+    NSString *insightsCacheKey = @"Insights";
+    NSString *todayCacheKey = @"Today";
+    
+    NSMutableDictionary *cacheDictionary = [self.ephemory objectForKey:cacheKey];
+    if (cacheDictionary.count == 3) {
```

Comment:
I'm going to be rewriting this method completely in another request and will follow the same pattern I did here:

https://github.com/wordpress-mobile/WordPressCom-Stats-iOS/blob/issue/254-insights/WordPressCom-Stats-iOS/WPStatsService.m#L96